### DISCUSSION TITLE
Unable to get a turtle renderer to work

### DISCUSSION THREAD

**Discussion by blockninja124** [cc-tweaked/CC-Tweaked#1836] [Q&A] [answered]
For my 1.19.2 CC:Tweaked addon I'm trying to register a turtle upgrade modeller. Since its not a flat item, I'm trying to use the `sided()` method. However, I can't get it to work. (I have managed to get the flat item renderer to work, but its not what I'm looking for). In the `FMLClientSetupEvent` I'm putting in 2 resource locations that should be pointing straight at my model file, but the turtle still renders as the purple and black cube. I don't really want to do any model generating like [advanced peripherals](https://github.com/IntelligenceModding/AdvancedPeripherals/blob/dev/1.20.1/src/main/java/de/srendi/advancedperipherals/common/data/RecipesProvider.java) does, which is also how [CC:Tweaked itself](https://github.com/cc-tweaked/CC-Tweaked/blob/f63f85921fb7ef27175df7bfe2f3c3d93475493b/projects/common/src/client/java/dan200/computercraft/client/ClientRegistry.java#L109) does it. So I thought I might have to, but I found the [ROSCC addon](https://github.com/purdue-arc/ROSCC/blob/1f6a5bc54164e7305c6c6fa76708edf960adaf86/src/main/java/com/purduearc/roscc/ROSCC.java#L132) that seems to do it with just a resource path to an existing model. I know that you can use a ModelResourceLocation instead of a ResourceLocation, but this seems to be for blockstates. Is it possible to register a renderer with a normal model location, and if so what am I doing wrong?

My model is under `assets/createturtles/models/block/assembler.json`
My client setup code:
```
public static void onRenderTypeSetup(FMLClientSetupEvent event) {
        //some other stuff

        ComputerCraftAPIClient.registerTurtleUpgradeModeller(ModTurtles.CONTRAPTION_TURTLE_UPGRADE.get(), TurtleUpgradeModeller.sided(new ResourceLocation(CreateTurtlesCore.MOD_ID, "block/assembler"), new ResourceLocation(CreateTurtlesCore.MOD_ID, "block/assembler"))); //I'll figure out the different sides later
    }
```

**Comment by SquidDev** [comment, marked as answer]
The issue here is that Minecraft doesn't load/bake every model by default — only those required by blockstates and items. This means that upgrade models typically won't be baked, and so fallback to the missing model. Later versions of CC:T will automatically [load these models](https://tweaked.cc/mc-1.20.x/javadoc/dan200/computercraft/api/client/turtle/TurtleUpgradeModeller.html#getDependencies()), but on 1.19.2 you need to specify them manually with `ModelEvent.RegisterAdditional`.

https://github.com/cc-tweaked/CC-Tweaked/blob/f4de575d35cc901b969f53ef82fe4d206f27237d/src/main/java/dan200/computercraft/client/ClientRegistry.java#L76-L83

**Reply by blockninja124** [reply]
Ah that worked! Thanks a lot. I guess I assumed that CC:Tweaked would take care of it, since I'm not having to register anything else that uses the model, but I also didn't really think about it. Its fair though that its not a feature in older versions, its kinda a specific thing to have in the API. But yeah, thanks!
